src/Client/Response.php

```diff
@@ -103,11 +92,14 @@ public function collect($key = null)
      * @param  bool  $sanitizeXmlFaultMessage
      * @return string
      */
-    public function body($transformXml = true, $sanitizeXmlFaultMessage = true)
+    public function body(bool $transformXml = true, bool $sanitizeXmlFaultMessage = true): string
     {
         $body = (string) $this->response->getBody();
         if ($transformXml && Str::contains($body, '<?xml')) {
-            $message = SoapXml::fromString($body)->getFaultMessage();
+            $message = Document::fromXmlString($body)
```

Comment:
Maybe better to use xpath()->evaluate() here

Reply:
you are reviewing old state 😄 
have that:
````php
            $message = Document::fromXmlString($body)
                    ->locate(elements_with_tagname('faultstring'))
                    ?->item(0)->firstChild->nodeValue
                ?? 'No Fault Message found';
````

Reply:
Not behind my computer this weekend. Here are some examples
https://github.com/veewee/xml/blob/master/docs/dom.md#evaluate

You can do something like `evaluate('string(/path/to/soap:faultString)', \Psl\Type\string())`, which will return the node value or an empty string if not found